### PR TITLE
Resolved defect whereby Clacks icon was always illuminated

### PR DESCRIPTION
**Pull request by SamuelCook**
In Chrome version 48, a change was made that affected the behaviour and appearance of all extensions. This resulted in the Clacks Overhead extension showing an illuminated icon on all pages, regardless of the presence of an `X-Clacks-Overhead` header or meta tag. This commit resolves this issue.

Prior to the release of Chrome 48, extensions could use the `chrome.pageAction.show` function to display an icon in the Chrome omnibox. The Clacks Overhead extension made use of this API to show an illuminated Clacks icon when an `X-Clacks-Overhead` response header or meta tag was detected in the active tab. Unless explicitly activated via this API call, Chrome would either omit the icon, or render it in a deactivated, greyed-out state.

Chrome 48 moved these icons to the Chrome toolbar, visible by default. Extension icons are now 'activated' - not greyed-out - for every page that the extension has the ability to access. It is no longer the case that an extension could inspect a page and programmatically decide whether or not to show its icon, nor to rely on Chrome to 'grey out' the icon in case of a lack of a call to `chrome.pageAction.show`.

These changes seem to have been made to communicate better the existence and activity of extensions in the browser, presumably to increase user awareness of what is running at a particular time. This is a probably a good change overall, but unfortunately one that has been communicated poorly, and has not been reflected in the developer documentation.

The best way to restore the previous behaviour seems to be to include a separate set of icons representing the 'greyed-out' state, and to alter the extension icon programmatically, rather than relying solely on `chrome.pageAction.show`.

Note that omitting the call to `chrome.pageAction.show` altogether results in the popup failing to render, so it is still necessary. It is just no longer sufficient.

For more information on the Chrome change that caused this defect, refer to the release notes at https://support.google.com/chrome/forum/AAAAP1KN0B0t9AHfd90OAE

The following discussions and defect reports may also be helpful:
https://bugs.chromium.org/p/chromium/issues/detail?id=597657
https://bugs.chromium.org/p/chromium/issues/detail?id=970908

### CODE REVIEW

**Comment by newfolder0** [comment]
Thank you very much for your advice! I will double check it in the next week or two and get back to you :).

**Comment by CacheMinimal** [comment]
Is there any way I can compile and test this prior to the merge?

<edit>
Found your fork over at https://github.com/SamuelCook/chrome-clacks and installed the code in Extension Developer mode.  All appears to work fine now.  I support the merge and publication to the Chrome Store.
</edit>

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 src.zip
 temp
+.idea
```
Comment (moorobaiken) [inline .gitignore]: Surely this should be ignored by your `.gitignore_global` file

Comment (SamuelCook) [reply .gitignore]: You're absolutely right. I'll remove this commit from the PR.

**Comment by egreer** [comment]
@moorobaiken Any status update on getting this merged and deployed to the webstore?  I am able to run this update locally as an unpacked extension on Chrome 78 and it appears to be working. 

Thank you both!

**Comment by moorobaiken** [comment]
@egreer I do not have the ability to merge tickets, you need to talk to @newfolder0

**Review by nhitze — APPROVED** [review]

**Comment by newfolder0** [comment]
Hi all, thanks for doing what you're doing and sorry I have dropped the ball.

I've written a more detailed comment on the original repo: https://github.com/newfolder0/chrome-clacks/issues/11#issuecomment-563003419

Let me know if you have any ideas to enable better collaboration, while still giving me the peace of mind that it is as safe and secure as possible to the users.

**Comment by newfolder0** [comment]
Thank you very much SamuelCook for this and others for your comments/review. Much appreciated, sorry for being a bit useless!

**Comment by newfolder0** [comment]
And I've uploaded the updated package to the Chrome Web Store. So it should auto-update for everyone who has installed it from there!

Thanks again, everyone :). Also for your patience.

**Pull request merged**